src/app/components/challenge/challengesubmit/challengesubmit.component.ts

```diff
@@ -394,12 +394,21 @@ export class ChallengesubmitComponent implements OnInit {
   formSubmit(self) {
     self.submissionError = '';
     const submissionFile = self.globalService.formItemForLabel(self.components, 'input_file').fileValue;
+    const submissionProjectUrl = self.globalService.formValueForLabel(self.components, 'project_url');
+    const submissionPublicationUrl = self.globalService.formValueForLabel(self.components, 'publication_url');
+    const regex = new RegExp('^(http[s]?:\\/\\/(www\\.)?|ftp:\\/\\/(www\\.)?|www\\.){1}([0-9A-Za-z-\\.@:%_\+~#=]+)+((\\.[a-zA-Z]{2,3})+)(/(.)*)?(\\?(.)*)?');
```

Comment:
Currently, it's more strict. That is, it's not allowing `myproj.github.io` but `http://myproj.github.io`. Can you please check?
Maybe you could try with the suggested change.

```suggestion
    const regex = new RegExp('[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)');
```

Reply:
I think my code works.  it's not allowing myproj.github.io but http://myproj.github.io

Reply:
And yours is not allowed anything

Reply:
Let's use this one? https://stackoverflow.com/questions/3809401/what-is-a-good-regular-expression-to-match-a-url

Regex Expression - `https?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b([-a-zA-Z0-9()@:%_\+.~#?&//=]*)`

Reply:
@robingan7 that was essentially my point. We should not want the user to explicitly type `http\\` or `https\\` before URL. This might make UX bad. When I had checked with the suggested change, it was allowing both the variants. But I'll encourage you to check on your end and there is no constraint to incorporate the suggested change. Please feel free to work-out on your own while ensuring that both variants are supported. Happy Hacking!! :smile:

Reply:
@lunayach I'm still confused. Do I allow the user to put in http\\ or https\\? and what are the `both variants`? Thanks

Reply:
@robingan7 we should allow the URL with (or without) http(s)\. Allowing and not allowing were `two variants`. I hope this clarifies. :smile: